### DISCUSSION TITLE
Localization

### DISCUSSION THREAD

**Discussion by Skeferstat** [spectreconsole/spectre.console#269] [Q&A] [answered]
Hi, can I use a localized text in the decription attribute from spectre.cli?

public class Settings : CommandSettings
  {
      [CommandOption("-n|--name <NAME>")]
      [Description("The person or thing to greet.")]          <----  [Description(Localization.MyText)]   
      [DefaultValue("World")]
      public string Name { get; set; }
  }
  Thanks for the answer!

**Comment by CyberSinh** [comment, marked as answer]
Hi,

Attributes only support constant values. But you can workaround this limitation like this:

```cs
internal sealed class LocalizedDescriptionAttribute : DescriptionAttribute
{
   public string Key { get; }

   public LocalizedDescriptionAttribute(string key) :
      base(SRCli.ResourceManager.GetString(key) ?? throw new ArgumentOutOfRangeException(nameof(key)))
   {
      Key = key;
   }
}

internal sealed class ConfigureCommandSettings : CommandSettings
{
    [CommandOption("--name")]
    [LocalizedDescription(nameof(SRCli.NameOption))]
    public string? Name { get; init; }
}
```

Hope this helps.
